File: HashTable/built_in_hash.cpp
int num = 3;
size_t hashNum = hash<int>()(num);
// Hash value of integer 3 is 3

bool bol = true;
size_t hashBol = hash<bool>()(bol);
// Hash value of boolean 1 is 1

double dec = 3.14159;
size_t hashDec = hash<double>()(dec);
// Hash value of decimal 3.14159 is 4614256650576692846

string str = "Hello 算法";
size_t hashStr = hash<string>()(str);
// Hash value of string "Hello 算法" is 15466937326284535026

// In C++, built-in std::hash() only provides hash values for basic data types
// Hash values for arrays and objects need to be implemented separately

File: java/hash_table/built_in_hash.java
int num = 3;
int hashNum = Integer.hashCode(num);
// Hash value of integer 3 is 3

boolean bol = true;
int hashBol = Boolean.hashCode(bol);
// Hash value of boolean true is 1231

double dec = 3.14159;
int hashDec = Double.hashCode(dec);
// Hash value of decimal 3.14159 is -1340954729

String str = "Hello 算法";
int hashStr = str.hashCode();
// Hash value of string "Hello 算法" is -727081396

Object[] arr = { 12836, "小哈" };
int hashTup = Arrays.hashCode(arr);
// Hash value of array [12836, 小哈] is 1151158

ListNode obj = new ListNode(0);
int hashObj = obj.hashCode();
// Hash value of ListNode object utils.ListNode@7dc5e7b4 is 2110121908
